### PR TITLE
Extend ViewOnly DAV plugin to versions endpoint

### PR DESCRIPTION
**Pull request by PVince81**
## Summary

Applies the ViewOnly DAV plugin also to the versions DAV endpoint

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/dav/lib/DAV/ViewOnlyPlugin.php [hunk 1]
```diff
@@ -70,13 +71,18 @@
...
+			if (!($davNode instanceof DavFile) && !($davNode instanceof VersionFile)) {
 				return true;
 			}
-			// Restrict view-only to nodes which are shared
-			$node = $davNode->getNode();
+
+			if ($davNode instanceof DavFile) {
+				// Restrict view-only to nodes which are shared
+				$node = $davNode->getNode();
+			} elseif ($davNode instanceof VersionFile) {
```
[the remark this answers is not in the document]
Comment (PVince81) [reply apps/dav/lib/DAV/ViewOnlyPlugin.php:81]: yeah yeah I know... I just thought it would be nice to see this explicitly

**Review by nickvergessen — APPROVED** [review]

**Review by juliusknorr — APPROVED** [review]

File: apps/dav/lib/DAV/ViewOnlyPlugin.php [hunk 2]
```diff
@@ -70,11 +71,16 @@ public function checkViewOnly(RequestInterface $request): bool {
...
 			}
-			// Restrict view-only to nodes which are shared
-			$node = $davNode->getNode();
+
+			if ($davNode instanceof DavFile) {
+				// Restrict view-only to nodes which are shared
+				$node = $davNode->getNode();
+			} else {
+				$node = $davNode->getVersion()->getSourceFile();
+			}
```
Comment (artonge) [inline apps/dav/lib/DAV/ViewOnlyPlugin.php:83]:
```suggestion
			} else if ($davNode instanceof VersionFile) {
				$node = $davNode->getVersion()->getSourceFile();
			} else {
				return true;
			}
```

Then you can drop the first condition and access to `VersionFile` methods is more explicit.

**Review by artonge — APPROVED** [review]

File: apps/dav/lib/DAV/ViewOnlyPlugin.php (hunk 2 above)
Comment (PVince81) [reply apps/dav/lib/DAV/ViewOnlyPlugin.php:83]: ok, I'll go with that. thanks a lot!

**Pull request merged**